server.js

```diff
@@ -3266,23 +3270,38 @@ cache(function(data, match, sendBadge, request) {
       var data = JSON.parse(buffer);
       var modifier = '';
       var issues;
-      if (isPR) {
-        issues = data.total_count;
+      if (isState && issuesApi) {
+        var rightSide = isRaw ? data.state : data.title;
+        rightSide = rightSide.length > 25 ? rightSide.substr(0,21)+'...' : rightSide
+        var leftSide = '#' + ghLabel;
+        if (!data.message) {
+          badgeData.colorscheme = (data.state == 'closed') ? 'red': 'brightgreen';
+        } else {
+          rightSide = data.message;
+          badgeData.colorscheme = 'lightgrey';
+        }
+        badgeData.text[0] = leftSide;
+        badgeData.text[1] = rightSide;
+        sendBadge(format, badgeData);
       } else {
-        if (issuesApi) {
-          issues = data.length;
-          if (res.headers['link'] &&
-              res.headers['link'].indexOf('rel="last"') >= 0) {
-            modifier = '+';
-          }
+        if (isPR) {
+          issues = data.total_count;
         } else {
-          issues = data.open_issues_count;
+          if (issuesApi) {
+            issues = data.length;
+            if (res.headers['link'] &&
+                res.headers['link'].indexOf('rel="last"') >= 0) {
+              modifier = '+';
+            }
+          } else {
+            issues = data.open_issues_count;
+          }
         }
+        var rightText = isRaw? '': (isClosed? ' closed': ' open');
```

Comment:
good call... i was typing fast to get this all done so we could host and add functionality